### DISCUSSION TITLE
Trouble getting output images

### DISCUSSION THREAD

**Discussion by clairewickenden** [joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#24] [Q&A]
Hi, thank you so much for creating this! It's exactly what I needed. I am having trouble resolving an issue I am getting when running the script. I have files in the input folder and when I run the script, it produces output images with file names correctly linked to my input folder files (so I know it's recognizing them at least). However, the output images are just the blank mockup template in jpg format. I am not sure why mine is having trouble actually embedding the input image.

I used the example scripts and edited to fit my files:

#include "script/Batch Mockup Smart Object Replacement.jsx" 

var output = {
  path: './_output files', // Outputs next to the mockup psd
  format: 'jpg', // 'jpg', 'png', 'tif', 'psd', 'pdf'
  filename: '@mockup - @input', 
};

mockups([
  
  // Mockup #1
  {
    output: output, 
    mockupPath: '$/assets/processed_mockups.psd', 
    smartObjects: [
      
      // Smart object #1
      {
        target: '@cwd-file',
        input: [
          './_input files'
        ], 
        inputNested: true,
        align: 'center center',
        resize: 'fill', 
      },
      
      // You could have more smart objects per mockup file...
      // {},
      
    ]
  },
  
  // You could have more mockup files...
  // {},
    
]);

and

// v.1.7.
// Batch Mockup Smart Object Replacement.jsx

// You'll need to incplude this file to another script file:
// #include "../script/Batch Mockup Smart Object Replacement.jsx" 
// Check the example folders for a few real world examples.

// Minimal settings
/*

#include "Batch Mockup Smart Object Replacement.jsx"

var outputOpts = {
  path: '$/_output' 
};

mockups([
  
  {
    output: outputOpts,
    mockupPath: '$/assets/processed_mockups.psd',
    smartObjects: [
      {
        target: '@cwd-file',
        input: '$/input',
      },
      // {..},  comma separate multiple smartobjects 
    ]
  },
  // {..},  comma separate multiple mockups 
  
]);

Also wanted to note my input files are PDF, if that makes a difference.
Thanks for any help!!
Claire

**Comment by joonaspaakko** [comment]
I kinda made the mistake of making the examples somewhat complex, so I usually recommend taking this [standalone example and](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement.js-Photoshop-script-standalone-example) and editing in your own files. It simplifies it a bit, maybe you could try that, if you didn't already. 

Honestly, I'm not sure if I ever tested pdf properly... I will take a look at that and your settings tomorrow, though often times it's hard to confirm anything without all the related files. 

If you could send me the project files it would help me recreate the issue... What comes in and out doesn't matter much visually, so if you want to keep your mockup or input files to yourself, you could include just one edited input file and perhaps downsized and flattened mockup file. Again, if privacy is important, you can email them straight to me, my email is in my profile page. Or send me a link if the size is too much.

**Reply by clairewickenden** [reply]
Thank you, I just sent you an email with my files!

**Reply by joonaspaakko** [reply]
@clairewickenden, found it! You have a mismatch with the mockup psd and the settings script. 

1. Your settings file says the smart object target is `target: '@cwd-file'`
2. The smart object layer name in the mockup .psd is `cwd`, so it can't find the target layer. 

> Seemed to work after correcting that.

**Reply by clairewickenden** [reply]
Oops, that was something I missed when switching to the standalone example. Thanks for looking into it... I just fixed that line of code on both jsx files but it is still outputting the blank images - same issue from before for me. Were you able to get it to work?

**Reply by joonaspaakko** [reply]
I see... Yes, it is working on my end. What Photoshop version are you using? And what operating system: Windows, Mac?

The script was tested in some of the more recent CC versions (2019-2023) and very lightly tested in CS 6, which did have some issues that I think were patched. Anything older than that and it likely won't work at all.

> Another good test would be to [download the standalone example](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement.js-Photoshop-script-standalone-example/archive/refs/heads/main.zip) (again) and run the settings script without changing anything. If it doesn't work (or any of the other examples in this main repository for that matter), then we can safely say it's not about any kind of user error. 

That said, I emailed you back the test I did, using your files, so you can try to run that as is.

**Reply by clairewickenden** [reply]
I am using Photoshop 2021 on a Mac. I must have had some other mistake or typos in my file because the one you emailed me worked perfectly. Thanks for your help!!
Claire
